mydumper.c

```diff
@@ -1796,7 +1796,7 @@ MYSQL_STMT *execute_detect_fields_stmt(MYSQL *conn, char *database, char *table,
 }
 
 gboolean detect_generated_fields(MYSQL *conn, char *database, char *table){
-	const char* query = "select 1 from information_schema.COLUMNS where TABLE_SCHEMA=? and TABLE_NAME=? and extra like '%GENERATED%'";
+	const char* query = "select 1 from information_schema.COLUMNS where TABLE_SCHEMA=? and TABLE_NAME=? and (extra like '%STORED GENERATED%' or extra like '%VIRTUAL GENERATED%')";
```

Comment:
Could we use `extra REGEXP '(STORED|VIRTUAL) GENERATED'` or even `extra = 'STORE GENERATED' or extra = 'VIRTUAL GENERATED'`?

Reply:
`REGEXP` is more simple, but I'm not sure whether it has a compatibility issue, I will check it.
`extra = 'STORE GENERATED' or extra = 'VIRTUAL GENERATED'` is not safe, maybe have other extra field

Reply:
I changed to use `regexp`.
`regexp` was introduced in [MariaDB 10.0.5](https://mariadb.com/kb/en/library/regexp/), which is quite an old version.
As for MySQL I found regexp in very early version 5.1, and I tested it on MySQL5.6 and MySQL5.7 all work well.
Btw I have tested this change on TiDB 2.0.0, TiDB3.0.0